Loop/Views/SimpleBolusView.swift

```diff
@@ -0,0 +1,349 @@
+                case .deliver:
+                    return Text("Deliver", comment: "Button text to deliver a bolus")
+                }
+            }
+        )
+        .buttonStyle(ActionButtonStyle(.primary))
+        .padding()
+    }
+    
+    private func alert(for alert: SimpleBolusViewModel.Alert) -> SwiftUI.Alert {
+        switch alert {
+        case .maxBolusExceeded:
+            guard let maximumBolusAmountString = viewModel.maximumBolusAmountString else {
+                fatalError("Impossible to exceed max bolus without a configured max bolus")
+            }
+            return SwiftUI.Alert(
+                title: Text("Exceeds Maximum Bolus", comment: "Alert title for a maximum bolus validation error"),
+                message: Text(String(format: NSLocalizedString("The maximum bolus amount is %1$@ U.", comment: "Format string for maximum bolus exceeded alert (1: maximumBolusAmount)"), maximumBolusAmountString))
+            )
+        case .carbEntryPersistenceFailure:
+            return SwiftUI.Alert(
+                title: Text("Unable to Save Carb Entry", comment: "Alert title for a carb entry persistence error"),
+                message: Text("An error occurred while trying to save your carb entry.", comment: "Alert message for a carb entry persistence error")
+            )
+        case .carbEntrySizeTooLarge:
+            let message = String(
+                format: NSLocalizedString("The maximum allowed amount is %1$@ grams", comment: "Alert body displayed for quantity greater than max (1: maximum quantity in grams)"),
+                NumberFormatter.localizedString(from: NSNumber(value: LoopConstants.maxCarbEntryQuantity.doubleValue(for: .gram())), number: .none)
+            )
+            return SwiftUI.Alert(
+                title: Text("Carb Entry Too Large", comment: "Alert title for a carb entry too large error"),
+                message: Text(message)
+            )
+        case .manualGlucoseEntryOutOfAcceptableRange:
+            let formatter = QuantityFormatter(for: viewModel.glucoseUnit)
+            let acceptableLowerBound = formatter.string(from: LoopConstants.validManualGlucoseEntryRange.lowerBound, for: viewModel.glucoseUnit) ?? String(describing: LoopConstants.validManualGlucoseEntryRange.lowerBound)
+            let acceptableUpperBound = formatter.string(from: LoopConstants.validManualGlucoseEntryRange.upperBound, for: viewModel.glucoseUnit) ?? String(describing: LoopConstants.validManualGlucoseEntryRange.upperBound)
+            return SwiftUI.Alert(
+                title: Text("Glucose Entry Out of Range", comment: "Alert title for a manual glucose entry out of range error"),
+                message: Text(String(format: NSLocalizedString("A manual glucose entry must be between %1$@ and %1$@", comment: "Alert message for a manual glucose entry out of range error. (1: acceptable lower bound) (2: acceptable upper bound)"), acceptableLowerBound, acceptableUpperBound))
```

Comment:
This looks like a typo...?

```suggestion
                message: Text(String(format: NSLocalizedString("A manual glucose entry must be between %1$@ and %2$@", comment: "Alert message for a manual glucose entry out of range error. (1: acceptable lower bound) (2: acceptable upper bound)"), acceptableLowerBound, acceptableUpperBound))
```

Reply:
Good catch!

Reply:
This still doesn't look fixed...?

Reply:
Doh. Thanks.